app/controllers/parent_objects_controller.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+class ParentObjectsController < ApplicationController
+  before_action :set_parent_object, only: [:show, :edit, :update, :destroy]
+
+  # GET /parent_objects
+  # GET /parent_objects.json
+  def index
+    @parent_objects = ParentObject.all
+  end
+
+  # GET /parent_objects/1
+  # GET /parent_objects/1.json
+  def show; end
+
+  # GET /parent_objects/new
+  def new
+    @parent_object = ParentObject.new
+  end
+
+  # GET /parent_objects/1/edit
+  def edit; end
+
+  def build_metadata_cloud_url(metadata_source, lb_record = "")
```

Comment:
this should be private or maybe a helper?

Reply:
on second thought if metadata_source is a model now. it should now how to make these urls for us.